main.star

```diff
@@ -70,25 +57,14 @@ def run(
         }
         ```
     """
+    snowflake_env = get_snowflake_env(snowflake_env)
+    aws_env = get_aws_env(aws_env) 
 
-    indexer_env_vars = get_snowflake_env(snowflake_env, is_running_in_ci)
-    if len(github_user_token) > 0:
-        indexer_env_vars["GITHUB_USER_TOKEN"] = github_user_token
-    else:
-        aws_env = get_aws_env(aws_env)
-        indexer_env_vars["AWS_ACCESS_KEY_ID"] = aws_env.access_key_id
-        indexer_env_vars["AWS_SECRET_ACCESS_KEY"] = aws_env.secret_access_key
-        indexer_env_vars["AWS_BUCKET_REGION"] = aws_env.bucket_region
-        indexer_env_vars["AWS_BUCKET_NAME"] = aws_env.bucket_name
-        indexer_env_vars["AWS_BUCKET_FOLDER"] = "{}/{}".format(aws_env.bucket_user_folder, AWS_S3_BUCKET_SUBFOLDER)
-
-    indexer_env_vars["LOGGER_LOG_LEVEL"] = DEFAULT_LOGGER_LOG_LEVEL
-
-    image_name_and_version = "{}:{}".format(KURTOSIS_PACKAGE_INDEXER_IMAGE, kurtosis_package_indexer_version)
     indexer_service = plan.add_service(
         name="kurtosis-package-indexer",
         config=ServiceConfig(
-            image=image_name_and_version,
+            # use local built docker image if no container image version specified
+            image=ImageBuildSpec(image_name="kurtosis-package-indexer", build_context_dir=".") if kurtosis_package_indexer_version == "" else "{}:{}".format(KURTOSIS_PACKAGE_INDEXER_IMAGE, kurtosis_package_indexer_version),
```

Comment:
added `ImageBuildSpec` so you can use the latest local when testing new local changes.